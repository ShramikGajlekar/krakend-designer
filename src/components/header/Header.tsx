import AppBar from '@material-ui/core/AppBar';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import React, { FunctionComponent } from 'react';

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            display: 'flex',
        },
        appBar: {
            zIndex: theme.zIndex.drawer + 1,
        },
    }),
);

export interface HeaderProps {
    title?: string;
    logoURL?: string;
}

export const Header: FunctionComponent<HeaderProps> = ({ title, logoURL }) => {
    const classes = useStyles();
    return (
        <>
            <AppBar style={{ margin: 0 }} className={classes.appBar} position="fixed">
                <Toolbar>
                    <Box my={1} pl={0} mx={0}>
                        {logoURL ? <img className="" src={logoURL} height="35px" alt="logo" /> : title}
                    </Box>
                    <div style={{ marginLeft: 'auto' }}>
                        <Button size="large" color="inherit">
                            Tab1
                        </Button>
                        <Button size="large" color="inherit">
                            Tab2
                        </Button>
                        <Button size="large" color="inherit">
                            Tab3
                        </Button>
                    </div>
                </Toolbar>
            </AppBar>
        </>
    );
};
