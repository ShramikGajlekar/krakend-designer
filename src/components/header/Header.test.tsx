import { render } from '@testing-library/react';
import React from 'react';
import { Header } from './Header';

describe('Header with default props', () => {
    it('match a snapshot', () => {
        const { container } = render(<Header />);
        expect(container).toMatchSnapshot();
    });
});

describe('Header with title & logo', () => {
    it('match a snapshot', () => {
        const { container } = render(<Header logoURL="/logo-white.png" title="Header" />);
        expect(container).toMatchSnapshot();
    });
});
