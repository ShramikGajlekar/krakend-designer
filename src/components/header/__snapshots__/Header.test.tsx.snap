// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with default props match a snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation4"
    style="margin: 0px;"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiBox-root MuiBox-root-3"
      />
      <div
        style="margin-left: auto;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Tab1
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Tab2
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Tab3
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header with title & logo match a snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-5 mui-fixed MuiPaper-elevation4"
    style="margin: 0px;"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <img
          alt="logo"
          class=""
          height="35px"
          src="/logo-white.png"
        />
      </div>
      <div
        style="margin-left: auto;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Tab1
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Tab2
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Tab3
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
